File: brentversal/src/main/java/com/brentversal/neighborhoodreview/entity/NeighborhoodReview.java
package com.brentversal.neighborhoodreview.entity;

// 이 클래스에서 사용할 Java/Spring/프로젝트 타입 불러옴
import com.brentversal.member.entity.Member;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@Entity
/*
  한 회원이 같은 동네에 여러 번 남길 수 있다.

  예전에는 (member_id, admin_code) 유니크 제약으로 한 명당 한 개만 두고 다시 쓰면 덮어썼는데,
  살면서 느낀 점이 시기마다 달라질 수 있어 여러 개를 남길 수 있게 바꿨다.

  주의: 이 제약을 지워도 Hibernate 의 ddl-auto=update 는 이미 만들어진 DB 제약을 지우지 않는다.
  운영 DB에서는 ALTER TABLE 로 직접 걷어내야 한다.
*/
@Table(name = "neighborhood_review")
// 회원이 특정 행정동에 작성한 한줄평과 수정시각을 DB에 저장하는 Entity임
public class NeighborhoodReview {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "review_id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = true)
    private Member member;

    @Column(name = "admin_code", length = 20, nullable = false)
    private String adminCode;

    @Column(name = "admin_name", length = 50, nullable = false)
    private String adminName;

    @Column(name = "district_name", length = 20, nullable = false)
    private String districtName;

    @Column(name = "content", length = 300, nullable = false)
    private String content;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}

File: frentversal/src/types/NeighborhoodReview.ts
/*
  동네 한줄평 타입.

  여기서 다루는 adminCode 는 "행정동" 코드다.
  동네 탐색(types/Neighborhood.ts)의 Neighborhood.id / dong 은 "법정동" 기준이라
  코드 체계가 서로 다르다. 두 값을 서로 바꿔 넣으면 안 된다.

  서버의 NeighborhoodReviewResponseDto 와 항목을 1:1 로 맞춰 두었다.
*/

// 한 사람이 같은 동네에 남길 수 있는 한줄평 길이 상한.
// 서버 엔티티의 content 컬럼 길이(300)와 같은 값이라 한쪽만 바꾸면 안 된다.
export const MAX_REVIEW_LENGTH = 300;

export type NeighborhoodReview = {
    id: number;
    // 작성자 표기. 서버가 보는 사람에 따라 다른 값을 채워 준다.
    // 일반 사용자·중개인·비로그인은 "익명", 관리자만 실제 이름을 받는다.
    memberName: string;
    adminCode: string;
    adminName: string;
    districtName: string;
    content: string;
    // 서버가 보내는 LocalDateTime 문자열 (예: "2026-08-20T11:44:00")
    updatedAt: string;
    // 로그인한 내가 쓴 글인지. true 인 항목에만 수정·삭제 버튼을 보여 준다.
    mine: boolean;
};

// 한줄평을 남길 때 보내는 값. 같은 동네에 다시 써도 덮어쓰지 않고 새 글로 쌓인다.
export type NeighborhoodReviewRequest = {
    adminCode: string;
    adminName: string;
    districtName: string;
    content: string;
};
